website/statistics.py

```diff
@@ -101,6 +104,106 @@ def get_class_stats(self, user, class_id):
+            user, class_id)
+        matrix_values = self.get_matrix_values(students, class_adventures_formatted, ticked_adventures, '1')
+
+        if not class_ or (class_["teacher"] != user["username"] and not is_admin(user)):
+            return utils.error_page(error=404, ui_message=gettext("no_such_class"))
+
+        return render_template(
+            "class-grid.html",
+            class_info={"id": class_id, "students": students, "name": class_["name"]},
+            current_page="grid_overview",
+            max_level=hedy.HEDY_MAX_LEVEL,
+            teacher_adventures=teacher_adventures,
+            class_adventures=class_adventures_formatted,
+            ticked_adventures=ticked_adventures,
+            matrix_values=matrix_values,
+            page_title=gettext("title_class grid_overview"),
+        )
+
+    @route("/grid_overview/class/<class_id>/level", methods=["GET"])
+    @requires_login
+    def change_dropdown_level(self, user, class_id):
+        level = request.args.get('level')
+        students, class_, teacher_adventures, class_adventures_formatted, ticked_adventures = self.get_grid_info(
+            user, class_id)
+        matrix_values = self.get_matrix_values(students, class_adventures_formatted, ticked_adventures, level)
+
+        return jinja_partials.render_partial("customize-grid/grid-levels.html",
+                                             level=level,
+                                             class_info={"id": class_id, "students": students, "name": class_["name"]},
+                                             current_page="grid_overview",
+                                             max_level=hedy.HEDY_MAX_LEVEL,
+                                             teacher_adventures=teacher_adventures,
+                                             class_adventures=class_adventures_formatted,
+                                             ticked_adventures=ticked_adventures,
+                                             matrix_values=matrix_values,
+                                             page_title=gettext("title_class grid_overview"),
+                                             )
+
+    @route("/grid_overview/class/<class_id>/level", methods=["POST"])
+    @requires_login
+    def change_checkbox(self, user, class_id):
+        level = request.args.get('level')
+        student_index = request.args.get('student_index', type=int)
+        adventure_index = request.args.get('adventure_index', type=int)
+        students, class_, teacher_adventures, class_adventures_formatted, ticked_adventures = self.get_grid_info(
+            user, class_id)
+        matrix_values = self.get_matrix_values(students, class_adventures_formatted, ticked_adventures, level)
+
+        for program in ticked_adventures[students[student_index]]:
+            if level and class_adventures_formatted[level][adventure_index] in program:
+                program_id = program[0]
```

Comment:
Does this need to be a numerical index? It makes the code a bit hard to read and a bit shaky. Can we do a dictionary instead so it is more clear what you are fetching?